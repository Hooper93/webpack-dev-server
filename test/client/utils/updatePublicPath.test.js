'use strict';

/* eslint-disable camelcase, no-undef */

describe('updatePublicPath', () => {
  const samples = [
    'test',
    'https://example.com',
    'https://example.com/path',
    'https://example.com/path/foo.js',
    'http://user:password@localhost/',
    'http://0.0.0.0',
    'https://localhost:123',
    'http://user:pass@[::]:8080',
    `http://0.0.0.0:9000?sockPath=${encodeURIComponent(
      '/path/to/sockjs-node'
    )}`,
    `http://0.0.0.0:9000?sockHost=${encodeURIComponent('my.host')}`,
    'http://0.0.0.0:9000?sockPort=8888',
    `http://0.0.0.0:9000?publicPath=${encodeURIComponent('/dist/')}`,
    `http://0.0.0.0:9000?publicPath=${encodeURIComponent('/long/dist/path/')}`,
    `http://0.0.0.0:9000?publicPath=${encodeURIComponent(
      'http://my.host:8888/dist/'
    )}`,
    `http://0.0.0.0:9000?sockPath=${encodeURIComponent(
      '/path/to/sockjs-node'
    )}&sockHost=${encodeURIComponent('my.host')}&sockPort=8888`,
    `http://0.0.0.0:9000?sockPath=${encodeURIComponent(
      '/path/to/sockjs-node'
    )}&sockHost=${encodeURIComponent(
      'my.host'
    )}&sockPort=8888&publicPath=${encodeURIComponent(
      'http://my.host:8888/dist/'
    )}`,
    `http://user:pass@[::]:8080?sockPath=${encodeURIComponent(
      '/path/to/sockjs-node'
    )}&sockPort=8888`,
    `http://user:pass@[::]:8080?sockPath=${encodeURIComponent(
      '/path/to/sockjs-node'
    )}&sockHost=${encodeURIComponent('my.host')}&sockPort=8888`,
  ];

  global.__webpack_public_path__ = '';

  samples.forEach((url) => {
    jest.doMock(
      '../../../client-src/default/utils/getCurrentScriptSource.js',
      () => () => url
    );

    const {
      default: updatePublicPath,
      // eslint-disable-next-line global-require
    } = require('../../../client-src/default/utils/updatePublicPath');

    test(`should set public path when __resourceQuery is ${url}`, () => {
      __webpack_public_path__ = '';
      updatePublicPath(`?${url}`);
      expect(__webpack_public_path__).toMatchSnapshot();
    });

    test(`should set public path when the current script source is ${url}`, () => {
      __webpack_public_path__ = '';
      updatePublicPath();
      expect(__webpack_public_path__).toMatchSnapshot();
    });

    jest.resetModules();
  });
});
