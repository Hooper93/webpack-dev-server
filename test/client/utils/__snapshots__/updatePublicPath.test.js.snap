// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updatePublicPath should set public path when __resourceQuery is http://0.0.0.0 1`] = `"http://localhost/"`;

exports[`updatePublicPath should set public path when __resourceQuery is http://0.0.0.0:9000?publicPath=%2Fdist%2F 1`] = `"http://localhost:9000/dist/"`;

exports[`updatePublicPath should set public path when __resourceQuery is http://0.0.0.0:9000?publicPath=%2Flong%2Fdist%2Fpath%2F 1`] = `"http://localhost:9000/long/dist/path/"`;

exports[`updatePublicPath should set public path when __resourceQuery is http://0.0.0.0:9000?publicPath=http%3A%2F%2Fmy.host%3A8888%2Fdist%2F 1`] = `"http://my.host:8888/dist/"`;

exports[`updatePublicPath should set public path when __resourceQuery is http://0.0.0.0:9000?sockHost=my.host 1`] = `"http://localhost:9000/"`;

exports[`updatePublicPath should set public path when __resourceQuery is http://0.0.0.0:9000?sockPath=%2Fpath%2Fto%2Fsockjs-node 1`] = `"http://localhost:9000/"`;

exports[`updatePublicPath should set public path when __resourceQuery is http://0.0.0.0:9000?sockPath=%2Fpath%2Fto%2Fsockjs-node&sockHost=my.host&sockPort=8888 1`] = `"http://localhost:9000/"`;

exports[`updatePublicPath should set public path when __resourceQuery is http://0.0.0.0:9000?sockPath=%2Fpath%2Fto%2Fsockjs-node&sockHost=my.host&sockPort=8888&publicPath=http%3A%2F%2Fmy.host%3A8888%2Fdist%2F 1`] = `"http://my.host:8888/dist/"`;

exports[`updatePublicPath should set public path when __resourceQuery is http://0.0.0.0:9000?sockPort=8888 1`] = `"http://localhost:9000/"`;

exports[`updatePublicPath should set public path when __resourceQuery is http://user:pass@[::]:8080 1`] = `"http://user:pass@localhost:8080/"`;

exports[`updatePublicPath should set public path when __resourceQuery is http://user:pass@[::]:8080?sockPath=%2Fpath%2Fto%2Fsockjs-node&sockHost=my.host&sockPort=8888 1`] = `"http://user:pass@localhost:8080/"`;

exports[`updatePublicPath should set public path when __resourceQuery is http://user:pass@[::]:8080?sockPath=%2Fpath%2Fto%2Fsockjs-node&sockPort=8888 1`] = `"http://user:pass@localhost:8080/"`;

exports[`updatePublicPath should set public path when __resourceQuery is http://user:password@localhost/ 1`] = `"http://user:password@localhost/"`;

exports[`updatePublicPath should set public path when __resourceQuery is https://example.com 1`] = `"https://example.com/"`;

exports[`updatePublicPath should set public path when __resourceQuery is https://example.com/path 1`] = `"https://example.com/"`;

exports[`updatePublicPath should set public path when __resourceQuery is https://example.com/path/foo.js 1`] = `"https://example.com/"`;

exports[`updatePublicPath should set public path when __resourceQuery is https://localhost:123 1`] = `"https://localhost:123/"`;

exports[`updatePublicPath should set public path when __resourceQuery is test 1`] = `"/"`;

exports[`updatePublicPath should set public path when the current script source is http://0.0.0.0 1`] = `"http://localhost/"`;

exports[`updatePublicPath should set public path when the current script source is http://0.0.0.0:9000?publicPath=%2Fdist%2F 1`] = `"http://localhost:9000/dist/"`;

exports[`updatePublicPath should set public path when the current script source is http://0.0.0.0:9000?publicPath=%2Flong%2Fdist%2Fpath%2F 1`] = `"http://localhost:9000/long/dist/path/"`;

exports[`updatePublicPath should set public path when the current script source is http://0.0.0.0:9000?publicPath=http%3A%2F%2Fmy.host%3A8888%2Fdist%2F 1`] = `"http://my.host:8888/dist/"`;

exports[`updatePublicPath should set public path when the current script source is http://0.0.0.0:9000?sockHost=my.host 1`] = `"http://localhost:9000/"`;

exports[`updatePublicPath should set public path when the current script source is http://0.0.0.0:9000?sockPath=%2Fpath%2Fto%2Fsockjs-node 1`] = `"http://localhost:9000/"`;

exports[`updatePublicPath should set public path when the current script source is http://0.0.0.0:9000?sockPath=%2Fpath%2Fto%2Fsockjs-node&sockHost=my.host&sockPort=8888 1`] = `"http://localhost:9000/"`;

exports[`updatePublicPath should set public path when the current script source is http://0.0.0.0:9000?sockPath=%2Fpath%2Fto%2Fsockjs-node&sockHost=my.host&sockPort=8888&publicPath=http%3A%2F%2Fmy.host%3A8888%2Fdist%2F 1`] = `"http://my.host:8888/dist/"`;

exports[`updatePublicPath should set public path when the current script source is http://0.0.0.0:9000?sockPort=8888 1`] = `"http://localhost:9000/"`;

exports[`updatePublicPath should set public path when the current script source is http://user:pass@[::]:8080 1`] = `"http://user:pass@localhost:8080/"`;

exports[`updatePublicPath should set public path when the current script source is http://user:pass@[::]:8080?sockPath=%2Fpath%2Fto%2Fsockjs-node&sockHost=my.host&sockPort=8888 1`] = `"http://user:pass@localhost:8080/"`;

exports[`updatePublicPath should set public path when the current script source is http://user:pass@[::]:8080?sockPath=%2Fpath%2Fto%2Fsockjs-node&sockPort=8888 1`] = `"http://user:pass@localhost:8080/"`;

exports[`updatePublicPath should set public path when the current script source is http://user:password@localhost/ 1`] = `"http://user:password@localhost/"`;

exports[`updatePublicPath should set public path when the current script source is https://example.com 1`] = `"https://example.com/"`;

exports[`updatePublicPath should set public path when the current script source is https://example.com/path 1`] = `"https://example.com/"`;

exports[`updatePublicPath should set public path when the current script source is https://example.com/path/foo.js 1`] = `"https://example.com/"`;

exports[`updatePublicPath should set public path when the current script source is https://localhost:123 1`] = `"https://localhost:123/"`;

exports[`updatePublicPath should set public path when the current script source is test 1`] = `"/"`;
